file/src/test/java/docs/javadsl/LogRotatorSinkTest.java

```diff
@@ -82,13 +112,30 @@ public static void main(String... args) {
     // #sample
     */
     Creator<Function<ByteString, Optional<Path>>> pathGeneratorCreator = timeBasedPathCreator;
+
+    Source<ByteString, NotUsed> source =
+        Source.from(Arrays.asList("test1", "test2", "test3", "test4", "test5", "test6"))
+            .map(ByteString::fromString);
     // #sample
     CompletionStage<Done> completion =
         Source.from(Arrays.asList("test1", "test2", "test3", "test4", "test5", "test6"))
             .map(ByteString::fromString)
             .runWith(LogRotatorSink.createFromFunction(pathGeneratorCreator), materializer);
+
+    // GZip compressing the data written
+    CompletionStage<Done> compressedCompletion =
+        source.runWith(
+            LogRotatorSink.withSinkFactory(
+                pathGeneratorCreator,
+                path ->
```

Comment:
I really like the API improvements. It is great to have the `path` reused that is returned from the `pathGeneratorCreator`.